### PR TITLE
Fix: #576 Disable the functionality to edit the default dashboard details

### LINKED ISSUE #576: [BUG] Disable the functionality to edit the default dashboard details
# Description
Since the details of default dashboards cannot be modified, we should deactivate this feature in the user interface.

# Steps to Reproduce
1. Go to any of the default dashboard page page  `/dashboard.html?id=a28f485c-4747-4024-bb6b-d230f101f852` and click on the gear icon on top right to edit the dashboard details.
2. Disable the functionality to edit the dashboard details since they are not editable with a small message/indication to the user that these details cannot be edited.

# Expected Behavior
- The ability to edit should be disabled only for default dashboards, without affecting the functionality of non-default dashboards.
- Rely on the API response to determine if these dashboards are editable. This approach simplifies the process of controlling which dashboards can be edited, as it only requires changes to the server code.

### PR DESCRIPTION
**Pull request by KiritoCyanPine**
# Description
Default dashboards are not editable, the changes make the input fields `readonly` and displays a bootstrap `tooltip` stating 'default dashboards cannot be modified'.

Fixes #576  

# Testing
Since the changes were made in javascript files, built the project and tested the UI.

Tests conducted
- Tested default dashboards
- Tested user-created dashboards
- Tested the dashboards on Mobile view similarly. (*optional)

default dashboard:
Screenshot 2024-03-01 at 1 18 40 AM

created dashboard:
Screenshot 2024-03-01 at 1 19 16 AM

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`. No go changes were involved.

### CODE REVIEW

File: static/js/dashboard.js [hunk 1]
```diff
@@ -1127,6 +1127,38 @@ function handleDbSettings() {
...
+        headers: {
+            'Content-Type': 'application/json; charset=utf-8',
+            'Accept': '*/*'
+        },
+        dataType: 'json',
+        crossDomain: true,
+    }).then(function (res) {
+        for (let [key, value] of Object.entries(res)) {
+            console.log(key, value)
+            if (key == dbId ){
```
Comment (nbhavana) [inline static/js/dashboard.js:1142]: remove the console log

Comment (KiritoCyanPine) [reply static/js/dashboard.js:1142]: removed the suggested Log statement.

**Review by nbhavana — COMMENTED** [review]
@sonamgupta21 can you take a look at this PR when you get a chance?

**Comment by nbhavana** [comment]
We do not need tooltip for this change. We just need to make the input non-editable when the user tries to edit the field.

**Comment by KiritoCyanPine** [comment]
> We do not need tooltip for this change. We just need to make the input non-editable when the user tries to edit the field.

understood, removed the tooltip functionality from PR.

**Review by sonamgupta21 — APPROVED** [review]
Looks good!

**Pull request merged**